extensions/resteasy-reactive/quarkus-resteasy-reactive-links/runtime/src/main/java/io/quarkus/resteasy/reactive/links/InjectRestLinks.java

```diff
@@ -0,0 +1,13 @@
+package io.quarkus.resteasy.reactive.links;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ ElementType.TYPE, ElementType.METHOD })
+public @interface InjectRestLinks {
+
+    RestLinkType value() default RestLinkType.TYPE;
```

Comment:
I wonder what purpose this serves. Can you show me use-cases? The example test could deduce that from the method return type.

Reply:
I thought that this would allow user to specify which links does he want to be injected.
E.g. if a `DELETE` operation returns a deleted entity, you don't necessarily want to return the links specific for that entity (self, update, delete) because it doesn't exist any more. But you might still want to return the links for getting all and creating new entities.

Reply:
OK I see, but that feels special-casy.